simulation/simulation.py

```diff
@@ -0,0 +1,295 @@
+            price = round(self.random_lot.uniform(0, 10), 2)
+            self.space_tasks.append(asyncio.ensure_future(space_routine(0,
+                                                                        px,
+                                                                        py,
+                                                                        n,
+                                                                        str(name),
+                                                                        price,
+                                                                        n,
+                                                                        self)))
+            count += n
+            name += 1
+
+        for i in range(self.no_cars):
+            self.car_tasks.append(asyncio.ensure_future(car_routine(round(self.random_car.uniform(0, self.max_time), 1),
+                                                                    self.random_car.randint(0, self.x-1),
+                                                                    self.random_car.randint(0, self.y-1),
+                                                                    self)))
+
+        self.tasks = self.car_tasks + self.space_tasks
+
+        framerate = 1/60
+        root = tk.Tk()
+        self.tasks.append(asyncio.ensure_future(self.run_tk(root, framerate)))
+        asyncio.get_event_loop().run_until_complete(asyncio.gather(*self.tasks))
+
+    @asyncio.coroutine
+    def run_tk(self, root, interval):
+        w = tk.Canvas(root, width=self.x, height=self.y)
+        w.pack()
+        try:
+            while True:
+                w.delete("all")
+                now = time.time()
+                for lot in self.lots:
+                    # TODO no need to redraw these every time
+                    lotlat, lotlong = lot.get_location()
+                    w.create_rectangle(lotlat, lotlong, lotlat + 20, lotlong + 20, width=0, fill="green")
+
+                for car in self.cars:
+                    if car.drawing:
+                        dotlat, dotlong = car.get_position(now)
+                        w.create_oval(dotlat, dotlong, dotlat + 5, dotlong + 5, width=0, fill='blue')
+                root.update()
+                yield from asyncio.sleep(interval)
+        except tk.TclError as e:
+            if "application has been destroyed" not in e.args[0]:
+                raise
+
+
+class Waypoint:
+    def __init__(self, timestamp, lat, long):
+        self.time = timestamp
+        self.lat = lat
+        self.long = long
+
+
+class Car:
+    def __init__(self, lat, long):
+        self.lat = lat
+        self.long = long
+        self.destX = 0
+        self.destY = 0
+        self.aDestX = 0
+        self.aDestY = 0
+        self.drawing = False
+        self.speed = 60
+        self.waypoints = []
+        self.waypoints.append(Waypoint(time.time(), self.lat, self.long))
+
+    def get_location(self):
```

Comment:
As with `ParkingLot` you don't need `get_` methods, you can just make `Car` store a `parking.shared.location.Location` object and store it in `self.location`